online-shop/src/app/login/login.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnInit } from '@angular/core';
+import {FormControl, FormGroup, Validators} from "@angular/forms";
+import { AuthService} from "../auth.service";
+import {Router, Routes} from "@angular/router";
+import { AppComponent} from "../app.component";
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+
+  loginForm: FormGroup = new FormGroup({
+    username: new FormControl(''),
+    password: new FormControl(''),
+
+  })
+
+  loggedInSuccessful = false;
+  loggInFail = false;
+
+  constructor(private auth: AuthService, private router: Router, private navigation: AppComponent) { }
+
+  ngOnInit(): void {
+  }
+
+  async submitLogin() {
+    const formValue = this.loginForm.value;
+    const login = await this.auth.login(formValue.username, formValue.password);
```

Comment:
**suggestion:** this await should not be here, an observable is not a promise. Either transform it using `firstValueFrom` or just use the subscribe